### ISSUE TITLE
강의: Stanford NLP: Lecture 10 | Recurrent Neural Networks

### ISSUE THREAD

**Issue by eubinecto** [eubinecto/k4ji_ai#15]
- https://www.youtube.com/watch?v=6niqTuYFZLQ

스탠포드에서는 RNN에 대해서 어떻게 설명하는지 한번 참고해보자!

**Comment by eubinecto** [comment]
핸즈온 머신 러닝에서 읽었던 것이 기억난다.

층이 매우 깊은 딥러닝 모델들의 vanishing & exploding gradient 문제를 해결하기 위해서 고안된 것이 batch normalisation이라고.

VGG16, VGG19가 그런 문제를 안고 있는 층이 매우 깊은 딥러닝 모델의 대표적인 예시.

이 두 무델이 만들어졌을 당시에는 batch normalisation이 고안되기 전이었다고 하는데, 그 때문에 모델이 converge 하지 않는 문제로 인해 여러가지 트릭이 필요했다고 함.

**Comment by eubinecto** [comment]
## Recurrent Neural Networks: Process sequences

<img width="1385" alt="image" src="https://user-images.githubusercontent.com/56193069/90359792-d44fc280-e094-11ea-9326-60969c1a80a8.png">

기존의 vanila neural network로는 one-to-one 만 가능.

RNN으로 sequence를 모델링을 할 수 있게 되면
- one-to-many (e.g. image captioning)
- many-to-many
- many-to-many (encoder-decoder)

도 가능해진다.

**Comment by eubinecto** [comment]
그렇게 RNN은 sequential data를 modeling하기 위해 만들어졌다. 하지만 non-sequential data를 모델링 하는데에도 RNN을 사용할 수있다. 오.. 흥미로운 접근.

## Sequential processing of Non-sequence data
<img width="1293" alt="image" src="https://user-images.githubusercontent.com/56193069/90360373-8c319f80-e096-11ea-9ea2-7586d48be80a.png">

이 예시는 digit classification을 many-to-one 으로 접근한 경우.  이미지는 sequential data는 아니기에 "many"에 적합하지는 않지만,  sequential data로 취급해서 digit classification을 진행하는 경우.

흥미로운 접근이다. 우리가 어떤 물체를 인식할 때, 한번의 glimpse로 알아볼 때도 있지만, 어떤 경우에는 물체의 여러부분을 여러 glimpses 로 살펴보아야 비로소 인식할 수 있는 경우도 있으니.


non-sequential data를 sequential data로 취급해서 분류문제를 접근한 경우. 

아래는 generation 문제를 접근한 경우.
<img width="1285" alt="image" src="https://user-images.githubusercontent.com/56193069/90360677-5214cd80-e097-11ea-909d-4289f50979d9.png">


"paint" the images one at time, sequentially.
